### ISSUE TITLE
Bombing combat prediction off

### ISSUE THREAD

**Issue by Der-Zorn-Gottes** [LoneGazebo/Community-Patch-DLL#1547]
The combat prediction for heavy bombers are wrong. If I attack a artilery, it predicts ~75% damage. Actually it does only ~25% damage.
Maybe it is not the prdiction that is wrong, because I would expect heavy bombers to wreck arties. And I noticed that my artilleries took very little damage from his bombers too.

**Comment by Iamblichos** [comment]
Is the artillery embarked or on land?

**Comment by ilteroi** [comment]
it's strange. the combat preview is generated from Lua via GetRangeCombatDamage(), but the actual combat is done via GetAirCombatDamage(). hence the difference.

ideally, Lua would rely on the same CombatInfo struct that is used for the actual combat, but it seems the preview is hand-crafted.

@LoneGazebo : grep for "UNLESS it's a boat" ... you changed this in one instance, but there are more. intended or not?

@all: what about GetAirDefenseDamage()? never thought about it before, but i don't think it makes sense. aircraft should take damage from interceptions and that's it. maybe a flat amount of damage for wear and tear after each mission, but none of this fake melee defender damages attacker stuff. opinions?

**Comment by Iamblichos** [comment]
sounds good in theory, but disparate tech levels would leave non air support armies defenseless, would really snowball the game for a tech leader I think.

**Comment by LoneGazebo** [comment]
But airplanes were and are game-changers in terms of military power.
@ilteroi , not sure, may need to make more changes for that. Make sure to merge into the latest (the last merge with your code was a chore, as it was a little behind!).

**Comment by Iamblichos** [comment]
Yeah, but they already are without having to do as he suggested.

**Comment by LoneGazebo** [comment]
fair enough, just noting the historical game-changer-ed-ness.

**Comment by ilteroi** [comment]
the idea would be to grant interception capability to additional unit types.

and maybe give 10hp wear and tear damage per air strike mission.

**Comment by LoneGazebo** [comment]
It makes sense to me, the question is whether or not that achieves an appreciably different game effect than the one we have now.

**Comment by ilteroi** [comment]
you're right. it probaby won't be game changing. but that's true for all late game changes. so why not do it?

**Comment by LoneGazebo** [comment]
I guess what I meant is, will it change the outcome of individual combats all that much? I'm fine with it, I just don't want you to put work in on it if you don't feel the outcome is worth the time-investment. That's all.

**Comment by ilteroi** [comment]
it's easy, no worries.

now back to the real issue. the lua issue is real, but it turns out that GetAirCombatDamage and GetRangedCombatDamage() are almost identical.

except for the bit you recently changed:

		//Correction - make this apply to all ranged units, naval too.
		else if (!pDefender->isRangedSupportFire() && pDefender->GetRangedCombatLimit() > 0)

which causes the defender to use RCS in one version and CS in the other.

what is GetRangedCombatLimit() about?

**Comment by LoneGazebo** [comment]
That should be GetRangedCombat() - is typo.

**Comment by WirelessKFC** [comment]
There are some issues with ranged combat too, when ranged land shoots another ranged land unit. the preview generates cs bonuses like moral for the defender but not rcs bonuses like barrage and accuracy. and i think they use cs when defending from ranged attacks rather than rcs.

Right now enemy crossbowman in citadel has 27.4 cs when I highlight it with lancer which is correct as 10cs * 274% = 27.4 (100% base + 100% terrain + 24% domination resist + 40% fortification + 10% bonus from moral) preview says 51 vs 17 major victory.

if I try to shoot the same enemy with a cuirassier preview only shows 21.25 cs the same bonuses are listed for the xbowman as before (but not 3 levels of barrage which affects rcs) as xbow has base rcs 20 i dont know how the number 21.25 is calculated. but the preview is predicting 72 damage. a lot more than the lancer... 

basically what Im seeing is ranged units doing massive damage against each other very often more than 50% in one shot. It's similar to the ranged ships one shotting each other problem

**Comment by WirelessKFC** [comment]
also ranged ships are using their cs to defend when getting shot by ranged land units resulting in them going down quickly

**Comment by ilteroi** [comment]
i think most of these effects are due to the typo mentioned above

**Comment by WirelessKFC** [comment]
what I mentioned is not only typo, but the actual damage is also consistent with the incorrect combat preview, basically ranged vs ranged is doing way too much damage against each other whether its land vs land, land vs sea or sea vs land, I think only sea vs sea was fixed
